frontend/js/src/search/UserSearch.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from "react";
+import { Link, useLoaderData, useNavigate } from "react-router-dom";
+import GlobalAppContext from "../utils/GlobalAppContext";
+
+type SearchResultsLoaderData = {
+  searchTerm: string;
+  users: [string, number, number?][];
+};
+
+export default function SearchResults() {
+  const navigate = useNavigate();
+  const { searchTerm, users } = useLoaderData() as SearchResultsLoaderData;
+  const { currentUser } = React.useContext(GlobalAppContext);
+
+  const [searchTermInput, setSearchTermInput] = React.useState(searchTerm);
+  const username = currentUser ? currentUser.name : null;
+
+  const search = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (!searchTermInput) {
+      return;
+    }
+    navigate(`/search/?search_term=${searchTermInput}`);
```

Comment:
Can we use useSearchParams here instead? https://reactrouter.com/en/main/hooks/use-search-params#usesearchparams

Reply:
Yep! It is a cleaner solution. Fixed